lib/implore.es6.js

```diff
@@ -144,6 +149,14 @@ export default function implore(request, callback) {
 				}
 			}
 
+			callback({
+				request,
+				response
+			});
+		} else if (xhr.readyState === 4 && xhr.status === 0) {
+			response.body = {};
+			response.status = 0;
```

Comment:
Hm, this isn't a semantically accurate default body/status code. In a time out scenario, the body is undefined, and status code is undefined right?

Likewise, with the existing fluxthis mplementation, an an empty object body only makes sense if the header type was application/json. I think we're making an assumption here that might cause issues for others 🤔

Reply:
If there's nothing obtainable from the body, perhaps it should just be left `undefined`?

Reply:
There may have been a race condition with my edit, so I'm not sure if you saw it! :)

I noticed that the new behaviour aligns with some of the existing error handling [behaviour](https://github.com/addthis/fluxthis/pull/106/files#diff-95a8310b9d902aabd8394a712a00caaaR146) - so perhaps these defaults align with the element of least surprise, i.e. it's consistent. But perhaps the existing behaviour is also surprising 🤔

As an aside; If we're suggesting there's an error associated with this scenario, do we need to attach an error to the response for consistency? 🤷‍♀️

Reply:
Is the responses status code of zero not enough of an indication?